# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../xdelta/gcc-4.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
--- xdelta-1.1.3/xdmain.c.orig	2005-09-11 16:50:35.568234060 +0200
+++ xdelta-1.1.3/xdmain.c	2005-09-11 16:50:49.562872061 +0200
@@ -64,7 +64,7 @@
 
 #include "xdelta.h"
 
-extern HandleFuncTable xd_handle_table;
+static HandleFuncTable xd_handle_table;
 
 #define XD_PAGE_SIZE (1<<20)
 
@@ -1541,7 +1541,7 @@
 
   xdp_source_add (gen, src);
 
-  if (! xd_handle_write (out, XDELTA_PREFIX, XDELTA_PREFIX_LEN))
+  if (! xd_handle_write (out, (guint8 *) XDELTA_PREFIX, XDELTA_PREFIX_LEN))
     return 2;
 
   /* compute the header */
@@ -1560,10 +1560,10 @@
   if (! xd_handle_write (out, (guint8*) header_space, HEADER_SPACE))
     return 2;
 
-  if (! xd_handle_write (out, from_name, strlen (from_name)))
+  if (! xd_handle_write (out, (guint8*) from_name, strlen (from_name)))
     return 2;
 
-  if (! xd_handle_write (out, to_name, strlen (to_name)))
+  if (! xd_handle_write (out,  (guint8*) to_name, strlen (to_name)))
     return 2;
 
   if (! xd_handle_close (out, 0))
@@ -1600,7 +1600,7 @@
   if (! xd_handle_putui (out, control_offset))
     return 2;
 
-  if (! xd_handle_write (out, XDELTA_PREFIX, XDELTA_PREFIX_LEN))
+  if (! xd_handle_write (out,  (guint8*) XDELTA_PREFIX, XDELTA_PREFIX_LEN))
     return 2;
 
   xd_read_close (from);
@@ -1637,7 +1637,7 @@
 
   ntohl_array (patch->header_space, HEADER_WORDS);
 
-  if (strncmp (patch->magic_buf, XDELTA_110_PREFIX, XDELTA_PREFIX_LEN) == 0)
+  if (strncmp ( patch->magic_buf, XDELTA_110_PREFIX, XDELTA_PREFIX_LEN) == 0)
     {
       patch->has_trailer = TRUE;
       patch->patch_version = "1.1";
